spec/facets/PartitionDatasetFacet.json

```diff
@@ -0,0 +1,63 @@
+{
+  "$schema" : "https://json-schema.org/draft/2020-12/schema",
+  "$id" : "https://openlineage.io/spec/facets/1-0-0/PartitionDatasetFacet.json",
+  "$defs" : {
+    "PartitionDatasetFacet" : {
+      "allOf" : [ {
+        "$ref" : "https://openlineage.io/spec/1-0-2/OpenLineage.json#/$defs/DatasetFacet"
+      }, {
+        "type" : "object",
+        "properties" : {
+          "columns": {
+            "type": "array",
+            "items": {
+              "type": "object",
+              "properties": {
+                "columnName": {
+                  "description": "Partitioning column name.",
+                  "type": "string",
+                  "example": "business_date"
+                }
+              }
+            }
+          },
+          "partitions": {
+            "type": "array", 
+            "items": {
+              "type": "object",
+              "properties": {
+                "identifier" : {
+                "description" : "List of partition keys and values describing a single partition.",
```

Comment:
Indenting here is off - I had a hard time parsing this.

Rather than wrapping this in an `identifier` with `column` and `value` properties, we could structure the partition records as just `column=value` properties. e.g., something like
```json
"partitions": {
    "type": "array",
    "items": {
        "type": "object",
        "additionalProperties": true
    }
}
```
which would allow you send items like
```json
{"partitions": [
    {"business_date": "2022-07-20"},
    {"countryCode": "US"}
]}
```

Reply:
are those the partitions that have been written by the run? If there are many this could be a bit verbose. Since the columns are captured above, maybe just an array of arrays?

Reply:
```
Rather than wrapping this in an identifier with column and value properties, we could structure the partition records as just column=value properties. e.g., something like
```

Unfortunately we lose static typing on the generated client side then.

Reply:
It would be generated as a `Map<String, String>`, right?